internal/adapters/terraform/aws/autoscaling/adapt.go

```diff
@@ -0,0 +1,143 @@
+	return LaunchConfig
+}
+
+func adaptAutoscaling(resource *terraform.Block, module *terraform.Module) autoscaling.AutoscalingGroupsList {
+	nameAttr := resource.GetAttribute("name")
+	nameVal := nameAttr.AsStringValueOrDefault("", resource)
+
+	var AZones []types.StringValue
+	AZAttr := resource.GetAttribute("availability_zones")
+	for _, AZ := range AZAttr.AsStringValues() {
+		AZones = append(AZones, AZ)
+	}
+
+	HCTAttr := resource.GetAttribute("health_check_type")
+	HCTVal := HCTAttr.AsStringValueOrDefault("", resource)
+
+	var LBNames []types.StringValue
+	LBNAttr := resource.GetAttribute("load_balancers")
+	for _, LBN := range LBNAttr.AsStringValues() {
+		LBNames = append(LBNames, LBN)
+	}
+
+	ASGArnAttr := resource.GetAttribute("arn")
+	ASGArnVal := ASGArnAttr.AsStringValueOrDefault("", resource)
+
+	DefaultCooldownAttr := resource.GetAttribute("default_cooldown")
+	DefaultCooldownVal := DefaultCooldownAttr.AsIntValueOrDefault(0, resource)
+
+	LaunchConfigurationNameAttr := resource.GetAttribute("launch_configuration")
+	LaunchConfigurationNameVal := LaunchConfigurationNameAttr.AsStringValueOrDefault("", resource)
+
+	var suspendedprocess []autoscaling.SuspendedProcesses
+	for _, susBlock := range resource.GetBlocks("suspended_processes") {
+
+		suspendedprocess = append(suspendedprocess, autoscaling.SuspendedProcesses{
+			Metadata: susBlock.GetMetadata(),
+		})
+	}
+
+	var Tags []autoscaling.Tags
+	tagsRes := resource.GetBlocks("tags")
+	for _, tagRes := range tagsRes {
+
+		Tags = append(Tags, autoscaling.Tags{
+			Metadata:   tagRes.GetMetadata(),
+			ResourceId: types.StringDefault("", tagRes.GetMetadata()),
+		})
+	}
+
+	return autoscaling.AutoscalingGroupsList{
+		Metadata:                resource.GetMetadata(),
+		Name:                    nameVal,
+		AvaiabilityZone:         AZones,
+		Instances:               nil,
+		HealthCheckType:         HCTVal,
+		LoadBalancerNames:       LBNames,
+		AutoScalingGroupARN:     ASGArnVal,
+		DefaultCooldown:         DefaultCooldownVal,
+		SuspendedProcesses:      suspendedprocess,
+		Tags:                    Tags,
+		LaunchConfigurationName: LaunchConfigurationNameVal,
+	}
+}
+
+func adaptNotificationConfiguration(resource *terraform.Block, module *terraform.Module) autoscaling.NotificationConfigurations {
+	asgnameAttr := resource.GetAttribute("group_names")
+	asgnameVal := asgnameAttr.AsStringValueOrDefault("", resource)
+
+	return autoscaling.NotificationConfigurations{
+		Metadata:             resource.GetMetadata(),
+		AutoScalingGroupName: asgnameVal,
+	}
+}
+
+func adaptLaunchConfiguration(resource *terraform.Block, module *terraform.Module) autoscaling.LaunchConfigurations {
+
+	imageIdAttr := resource.GetAttribute("image_id")
+	imageIdVal := imageIdAttr.AsStringValueOrDefault("", resource)
+
+	userDataAttr := resource.GetAttribute("user_data")
+	userDataVal := userDataAttr.AsStringValueOrDefault("", resource)
+
+	iamInstanceProfileAttr := resource.GetAttribute(("iam_instance_profile"))
```

Comment:
redundant parenthesis?
```suggestion
	iamInstanceProfileAttr := resource.GetAttribute("iam_instance_profile")
```

Reply:
done

> I see you tried to resolve some issues @realwebdev but did you forget to push your changes? I don't see any new commits.

Yes I haven't pushed the code. will test it then push the changes